### ISSUE TITLE
message packets need a new structure

### ISSUE THREAD

**Issue by JacopoWolf** [JacopoWolf/PotatoChatProtocol#17]
looking at it I've begun to think that message classes need to inherit from a major abstract APCPMessagePacket class (A is abstract).
The structure would be as following
```
  IPCPpacket
      ↑
AbsMessagePacket
↑
|- UserToUser
|- UserToGroup
|- Recieved

```

**Comment by gfurri20** [comment]
I think could be a good idea, we discuss it better on Friday

**Comment by JacopoWolf** [comment]
fixed with 2a4795c
